source/initial_conditions/harmonic_perturbation.cc

```diff
@@ -84,7 +90,19 @@ namespace aspect
                        ExcMessage ("Spherical harmonics can only be computed for "
                                    "degree >= 0."));
               // use a spherical harmonic function as lateral perturbation
+#ifdef DEBUG
+#ifdef ASPECT_USE_FP_EXCEPTIONS
+  // disable floating point exceptions for the boost call below
+  feclearexcept(FE_DIVBYZERO|FE_INVALID);
+#endif
+#endif
               lateral_perturbation = boost::math::spherical_harmonic_r(lateral_wave_number_1,lateral_wave_number_2,scoord[2],scoord[1]);
+#ifdef DEBUG
+#ifdef ASPECT_USE_FP_EXCEPTIONS
+  // enable floating point exceptions again
+  feenableexcept(FE_DIVBYZERO|FE_INVALID);
```

Comment:
I think it would be easier to maintain in the long run if you had a call to `fegetexcept` before the `feclearexcept`, and then simply restore things afterwards via `fesetexcept`. You could even do this without all of the preprocessor directives.